gravitee-apim-console-webui/src/management/application/details/subscriptions/components/subscription-api-keys/subscription-api-keys.component.ts

```diff
@@ -0,0 +1,209 @@
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { CommonModule } from '@angular/common';
+import { ChangeDetectionStrategy, Component, DestroyRef, inject, Input, OnChanges } from '@angular/core';
+import { MatButtonModule } from '@angular/material/button';
+import { MatTableModule } from '@angular/material/table';
+import { MatTooltipModule } from '@angular/material/tooltip';
+import {
+  GIO_DIALOG_WIDTH,
+  GioClipboardModule,
+  GioConfirmDialogComponent,
+  GioConfirmDialogData,
+  GioIconsModule,
+  GioLoaderModule,
+} from '@gravitee/ui-particles-angular';
+import { MatCardModule } from '@angular/material/card';
+import { filter, map, startWith, switchMap } from 'rxjs/operators';
+import { MatDialog } from '@angular/material/dialog';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+import { MatFormFieldModule } from '@angular/material/form-field';
+import { MatInputModule } from '@angular/material/input';
+import { MatSortModule } from '@angular/material/sort';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { isEqual } from 'lodash';
+
+import { gioTableFilterCollection } from '../../../../../../shared/components/gio-table-wrapper/gio-table-wrapper.util';
+import { ApplicationSubscriptionService } from '../../../../../../services-ngx/application-subscription.service';
+import { SnackBarService } from '../../../../../../services-ngx/snack-bar.service';
+import { GioPermissionModule } from '../../../../../../shared/components/gio-permission/gio-permission.module';
+import { GioTableWrapperFilters } from '../../../../../../shared/components/gio-table-wrapper/gio-table-wrapper.component';
+import { GioTableWrapperModule } from '../../../../../../shared/components/gio-table-wrapper/gio-table-wrapper.module';
+
+type ApiKeyVM = {
+  id: string;
+  key: string;
+  createdAt: number;
+  endDate: number;
+  isValid: boolean;
+};
+@Component({
+  selector: 'subscription-api-keys',
+  templateUrl: './subscription-api-keys.component.html',
+  styleUrls: ['./subscription-api-keys.component.scss'],
+  imports: [
+    CommonModule,
+    GioIconsModule,
+    GioLoaderModule,
+    MatButtonModule,
+    MatTableModule,
+    MatTooltipModule,
+    MatCardModule,
+    MatFormFieldModule,
+    MatInputModule,
+    MatSortModule,
+    GioIconsModule,
+    GioTableWrapperModule,
+    GioClipboardModule,
+    GioPermissionModule,
+  ],
+  standalone: true,
+  changeDetection: ChangeDetectionStrategy.OnPush,
```

Comment:
Why do we need this change detection stategy here ? It's not an issue I'm just curious 😄

Reply:
because it's a good practice. I'm going to try to add it to all the components I'm going to work on.